modules/trusted-firmware-m/interface/interface.c

```diff
@@ -24,6 +24,10 @@
  */
 K_MUTEX_DEFINE(tfm_mutex);
 
+
+int z_arm_save_fp_context(uint32_t *buffer, size_t buffer_size_words, bool *context_saved);
+void z_arm_restore_fp_context(uint32_t *buffer, bool context_saved);
```

Comment:
If reusability is the goal here, these should probably go somewhere into `include/arch/arm/aarch32/` (maybe `exc.h`?).

Also, it would be nice to refactor "FP context" struct into its own and use it here:
https://github.com/zephyrproject-rtos/zephyr/blob/21d1ad3762302b3e461953df59430c77e0709274/include/arch/arm/aarch32/exc.h#L97-L98

Reply:
Right. I was thinking that we could decide on adding the declarations, once we actually have the re-use case in some "in-tree" use-case. So I felt it is ok for now to keep the declarations where the functions are used. But sure, we could declare of course the functions in this arch file

Reply:
>Also, it would be nice to refactor "FP context" struct into its own and use it here:

The context is not quite the same. In exc.h, only 16 of the 32 fp registers are in the struct (the callee saved registers). But I can make an analogous struct.